src/main/java/org/sopt/service/PostService.java

```diff
@@ -0,0 +1,77 @@
+package org.sopt.service;
+
+import static org.sopt.service.validator.PostValidator.*;
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.sopt.domain.Post;
+import org.sopt.exception.DuplicateTitleException;
+import org.sopt.global.CheckTime;
+import org.sopt.repository.PostRepository;
+
+public class PostService {
+
+	private final PostRepository postRepository = new PostRepository();
+
+
+	public void createPost(Post post){
+		if(!findDuplicateTitle(post.getTitle(),mapToList())){
+			postRepository.save(post);
+			CheckTime.setTimestamp();
+		}else {
+			throw new DuplicateTitleException();
+		}
+	}
+
+	public List<Post> getAllPosts(){
+		return mapToList();
+	}
+
+	public Post getPostById(int id){
+		Post post = postRepository.findById(id);
+		if(post == null){
+			return null;
+		}
+		return postRepository.findById(id);
```

Comment:
null 값에 대한 처리는 항상 신중하게 할 필요가 있습니다 ! 자바 내에서 NPE에 대한 처리는 항상 골치아프거든요.

현재는 프로젝트의 크기가 크지 않기에 이정도여도 충분할 것 같습니다.

다만, 다음과 같은 방향으로 코드를 조금 더 명확하게 보일 수 있을 것 같아요 !

```suggestion
return postRepository.findById(id);
```

위의 코드 한 줄만 있어도 충분할 것 같습니다 !

Reply:
감사합니다 ! 한 줄로 작성할 코드를 나눠서 작성해두었었네요!